packages/notifi-react-card/lib/components/subscription/NotifiSubscriptionCard.tsx

```diff
@@ -110,6 +112,16 @@ export const NotifiSubscriptionCard: React.FC<
 
   let contents: React.ReactNode = null;
 
+  window.onfocus = () => {
+    // Ensure target is up-to-date after user returns to tab from 3rd party verification site
+    if (!isClientInitialized || !isAuthenticated) return;
+
+    if (canaryIsActive) {
+      return frontendClient.fetchData().then(render);
+    }
+    reload();
+  };
+
```

Comment:
we should install side-effects only inside `useEffect` hooks

```
useEffect(() => {
  const handler = () => {
    if (!isClientInitialized || !isAuthenticated) {
      return;
    }
    if (canaryIsActive) {
      return frontendClient.fetchData().then(render);
    }
    reload();
  };
  
  window.addEventListener('focus', handler);
  return () => {
    window.removeEventListener('focus', handler);
  };
}, [isClientInitialized, isAuthenticated, canaryIsActive, frontendClient]);

Reply:
Thanks! It is much better solution!!